ui/job-view/details/BugFiler.jsx

```diff
@@ -0,0 +1,610 @@
+      if (productResp.ok) {
+        const productObject = productData.products[0];
+        // Find the newest version for the product that is_active
+        const version = productObject.versions.filter(prodVer => prodVer.is_active).slice(-1)[0];
+        const payload = {
+          product,
+          component,
+          summary,
+          keywords,
+          version: version.name,
+          blocks,
+          depends_on: dependsOn,
+          see_also: seeAlso,
+          crash_signature: crashSignature,
+          severity,
+          priority,
+          comment: descriptionStrings,
+          comment_tags: 'treeherder',
+        };
+
+        const bugResp = await create(getApiUrl('/bugzilla/create_bug/'), payload);
+        // const bugResp = await create('http://httpstat.us/404', payload);
+        const data = await bugResp.json();
+        if (bugResp.ok) {
+          successCallback(data);
+          toggle();
+        } else {
+          this.submitFailure('Treeherder Bug Filer API', bugResp.status, bugResp.statusText, data);
+        }
+      } else {
+        this.submitFailure('Bugzilla', productResp.status, productResp.statusText, productData);
+      }
+    } catch (e) {
+      notify.send(`Error filing bug: ${e.toString()}`, 'danger', { sticky: true });
+    }
+  }
+
+  submitFailure(source, status, statusText, data) {
+    const { notify } = this.props;
+
+    let failureString = `${source} returned status ${status}(${statusText})`;
+    if (data && data.failure) {
+      failureString += '\n\n' + data.failure;
+    }
+    if (status === 403) {
+      failureString += '\n\nAuthentication failed. Has your Treeherder session expired?';
+    }
+    notify.send(failureString, 'danger', { sticky: true });
+  }
+
+  toggleTooltip(key) {
+    const { tooltipOpen } = this.state;
+    this.setState({ tooltipOpen: { ...tooltipOpen, [key]: !tooltipOpen[key] } });
+  }
+
+  render() {
+    const {
+      isOpen, toggle, suggestion, parsedLog, fullLog, reftestUrl,
+    } = this.props;
+    const {
+      productSearch, suggestedProducts, thisFailure, isFilerSummaryVisible,
+      isIntermittent, summary, searching, checkedLogLinks, tooltipOpen,
+      selectedProduct,
+    } = this.state;
+    const searchTerms = suggestion.search_terms;
+    const crash = summary.match(crashRegex);
+    const crashSignatures = crash ? [crash[0].split('application crashed ')[1]] : [];
+    const unhelpfulSummaryReason = this.getUnhelpfulSummaryReason(summary);
+
+    return (
+      <div>
+        <Modal isOpen={isOpen} toggle={toggle} size="lg">
+          <ModalHeader toggle={toggle}>Intermittent Bug Filer</ModalHeader>
+          <ModalBody>
+            <form id="modalForm">
+              <input
+                name="modalProductFinderSearch"
+                id="modalProductFinderSearch"
+                onKeyDown={this.productSearchEnter}
+                onChange={evt => this.setState({ productSearch: evt.target.value })}
```

Comment:
I thought calling setState in render was problematic due to the state/render cycle? Is that not the case if its in-lined in jsx? (seems to work fine when testing)